x-pack/platform/test/saved_object_api_integration/low-level/apis/find.ts

```diff
@@ -0,0 +1,305 @@
+        expect(idResponse.body.message).to.match(/Fielddata access on the _id field is disallowed/);
+
+        const scoreQuery = queryString.stringify({
+          type: sortTestTypes,
+          sort_field: sortTestFields.score,
+        });
+        const scoreResponse = await supertest.get(
+          `${getUrlPrefix(defaultNamespace)}/api/saved_objects/_find?${scoreQuery}`
+        );
+        expect(scoreResponse.status).to.eql(200, JSON.stringify(scoreResponse.body));
+      });
+
+      it('throws error if sort field is unknown', async () => {
+        const query = queryString.stringify({
+          type: 'sortTestingType',
+          sort_field: 'unknownField',
+        });
+        const response = await supertest.get(
+          `${getUrlPrefix(defaultNamespace)}/api/saved_objects/_find?${query}`
+        );
+        expect(response.status).to.eql(400);
+        expect(response.body.message).to.match(/Unknown sort field unknownField/);
+      });
+
+      it('sorts by textWithKeyword.keyword for single type', async () => {
+        const query = queryString.stringify({
+          type: 'sortTestingType',
+          sort_field: 'textWithKeyword.keyword',
+          sort_order: 'desc',
+        });
+        const response = await supertest.get(
+          `${getUrlPrefix(defaultNamespace)}/api/saved_objects/_find?${query}`
+        );
+        expect(response.status).to.eql(200, JSON.stringify(response.body));
+        expect(response.body.saved_objects.map((o: any) => o.attributes.textWithKeyword)).to.eql([
+          'zulu',
+          'yankee',
+          'victor',
+        ]);
+      });
+
+      it('sorts by textWithKeyword.keyword for multiple types', async () => {
+        const query = queryString.stringify({
+          type: sortTestTypes,
+          sort_field: 'textWithKeyword.keyword',
+          sort_order: 'desc',
+        });
+        const response = await supertest.get(
+          `${getUrlPrefix(defaultNamespace)}/api/saved_objects/_find?${query}`
+        );
+        expect(response.status).to.eql(200, JSON.stringify(response.body));
+        expect(response.body.saved_objects.map((o: any) => o.attributes.textWithKeyword)).to.eql([
+          'zulu2',
+          'zulu',
+          'yankee2',
+          'yankee',
+          'victor2',
+          'victor',
+        ]);
+      });
+
+      it('throws error if sort field is not present in single type', async () => {
+        const query = queryString.stringify({
+          type: 'sortTestingType',
+          sort_field: 'notPresent',
+        });
+        const response = await supertest.get(
+          `${getUrlPrefix(defaultNamespace)}/api/saved_objects/_find?${query}`
+        );
+        expect(response.status).to.eql(400);
+        expect(response.body.message).to.match(/Unknown sort field notPresent/);
+      });
+
+      it('sorts by keyword field for multiple types when some documents are missing the field', async () => {
```

Comment:
Can you add a end-to-end test for sorting on a short + long field to show different numeric fields are treated equally